src/registry.rs

```diff
@@ -266,8 +266,52 @@ pub enum NpmRegistryPackageInfoLoadError {
   LoadError(#[from] Arc<anyhow::Error>),
 }
 
+/// A trait for getting package information from the npm registry, with no
+/// internal caching mechanism.
+#[async_trait]
+pub trait NpmRegistryApiWithoutCache: Sync + Send {
```

Comment:
Perhaps we could just have `NpmRegistryApi` to have default implementations that don't offer caching so we don't need an extra trait?

Reply:
Good idea!